agent-control/src/cli/on_host/host_monitoring_gen/infra_config.rs

```diff
@@ -0,0 +1,254 @@
+pub const INFRA_AGENT_TYPE_VERSION: &str = "newrelic/com.newrelic.infrastructure:0.1.0";
+
+/// Represents the values to create or migrate an infra-config
+pub struct InfraConfig {
+    values: HashMap<String, serde_yaml::Value>,
+    deletions: Vec<serde_yaml::Value>,
+}
+
+impl Default for InfraConfig {
+    fn default() -> InfraConfig {
+        Self {
+            values: HashMap::from([
+                (
+                    "license_key".to_string(),
+                    serde_yaml::Value::String("{{NEW_RELIC_LICENSE_KEY}}".to_string()),
+                ),
+                (
+                    "enable_process_metrics".to_string(),
+                    serde_yaml::Value::Bool(true),
+                ),
+                (
+                    "status_server_enabled".to_string(),
+                    serde_yaml::Value::Bool(true),
+                ),
+                (
+                    "status_server_port".to_string(),
+                    serde_yaml::Value::Number(18003.into()),
+                ),
+            ]),
+            deletions: vec![
+                serde_yaml::Value::String("staging".to_string()),
+                serde_yaml::Value::String("enable_process_metrics".to_string()),
+                serde_yaml::Value::String("status_server_enabled".to_string()),
+                serde_yaml::Value::String("status_server_port".to_string()),
+                serde_yaml::Value::String("license_key".to_string()),
+                serde_yaml::Value::String("custom_attributes".to_string()),
+                serde_yaml::Value::String("is_integrations_only".to_string()),
+            ],
+        }
+    }
+}
+
+impl InfraConfig {
+    pub fn with_custom_attributes(mut self, custom_attributes: &str) -> Result<Self, CliError> {
+        if !custom_attributes.trim().is_empty() {
+            let custom_attributes_value: serde_yaml::Value =
+                serde_yaml::from_str(custom_attributes).map_err(|err| {
+                    CliError::Command(format!("error parsing custom attributes: {err}"))
+                })?;
+            if let Some(attributes) = custom_attributes_value.as_mapping() {
+                for (key, value) in attributes {
+                    key.as_str()
+                        .and_then(|key| self.values.insert(key.to_string(), value.clone()));
+                }
+            }
+        }
+        Ok(self)
+    }
+
+    pub fn with_region(mut self, region: Region) -> Self {
+        if region == Region::STAGING {
+            self.values
+                .insert("staging".to_string(), serde_yaml::Value::Bool(true));
+        }
+        self
+    }
+
+    pub fn with_proxy(mut self, proxy: &str) -> Self {
+        if !proxy.trim().is_empty() {
+            self.deletions
+                .push(serde_yaml::Value::String("proxy".to_string()));
+            self.values.insert(
+                "proxy".to_string(),
+                serde_yaml::Value::String(proxy.to_string()),
+            );
```

Comment:
Why is it added to both arrays?

Reply:
Because we want to delete the previous appearance and overwrite it with a new one.